crates/nu-command/src/formats/to/nuon.rs

```diff
@@ -189,26 +196,29 @@ fn to_nuon(call: &Call, input: PipelineData) -> Result<String, ShellError> {
     value_to_string(&v, call.head)
 }
 
+// This hits:
+// • Any character of []:`{}#"';()|$,
+// • Any digit (\d)
+// • Any whitespace (\s)
+static NEEDS_QUOTES_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r#"[\[\]:`\{\}#'";\(\)\|\$,\s\d]"#)
+        .expect("internal error: NEEDS_QUOTES_REGEX didn't compile")
+});
+
 fn needs_quotes(string: &str) -> bool {
-    string.contains(' ')
-        || string.contains('[')
-        || string.contains(']')
-        || string.contains(':')
-        || string.contains('`')
-        || string.contains('{')
-        || string.contains('}')
-        || string.contains('#')
-        || string.contains('\'')
-        || string.contains(';')
-        || string.contains('(')
-        || string.contains(')')
-        || string.contains('|')
-        || string.contains('$')
-        || string.contains(',')
-        || string.contains('\t')
-        || string.contains('\n')
-        || string.contains('\r')
-        || string.contains('\"')
+    // These are case-sensitive keywords
+    match string {
+        "true" | "false" | "null" => return true,
+        _ => (),
+    };
+    // These are case-insensitive keywords
+    match string.to_lowercase().as_str() {
+        "inf" | "infinity" | "nan" | "+inf" | "+infinity" | "+nan" | "-inf" | "-infinity"
+        | "-nan" => return true,
+        _ => (),
```

Comment:
This allocates unnecessarily (especially for long strings that would be rejected due to spaces). 
Either a regex due to shared substrings or simpler https://doc.rust-lang.org/stable/std/primitive.str.html#method.eq_ignore_ascii_case

Reply:
OK, I've moved the inf/infinity/nan cases back into NEEDS_QUOTES_REGEX.